File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n= height.size();
     int left =0, right = n-1;
     int ans=0;

     while(left<right){
        int length = right - left;
        int height1 = min(height[left], height[right]);
        int area = length*height1;
        ans = max(ans,area);
        if(height[left]<height[right]){
            left++;
        }
        else{
            right--;
        }
     } 
     return ans;  
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n= height.length;
        int lp = 0, rp = n-1, maxWater = 0;
        while(lp < rp){
            int w = rp - lp;
            int ht = Math.min(height[lp], height[rp]);
            int areaWater = w*ht;
            maxWater = Math.max(maxWater, areaWater); 

        if(height[lp] < height[rp])
        lp++ ;
        else
        rp--;
        }
        return maxWater;
    }
}